File: 0049-group-anagrams/0049-group-anagrams.py
from collections import defaultdict
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        anag_dict=defaultdict(list)
        for s in strs:
            count=[0]*26
            for char in s:
                count[ord(char)-ord("a")]+=1
            key=tuple(count)
            anag_dict[key].append(s)
        return anag_dict.values()

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> hashmap;
        vector<vector<string>> result;
        for(int i=0;i<strs.size();i++){
            string ss;
            ss=strs[i];
            sort(ss.begin(),ss.end());
            hashmap[ss].push_back(strs[i]);
        }
        for(auto& entry:hashmap){
            result.push_back(entry.second);
        }
        return result;
    }
};
